airsonic-main/src/main/java/org/airsonic/player/controller/HomeController.java

```diff
@@ -114,6 +114,7 @@ protected ModelAndView handleRequestInternal(HttpServletRequest request) throws
                 map.put("genres", genres);
                 if (!genres.isEmpty()) {
                     String genre = getStringParameter(request, "genre", genres.get(0).getName());
+                    genre = genre.replaceAll("[^\\p{IsAlphabetic}^\\p{IsDigit}]", "");
```

Comment:
This breaks things for some genres (e.g. `Heavy Metal` has a space in it, likely others).

Reply:
You're right, I should use proper escaping instead of a ghetto-regex.